public-dashboard-client/src/controls/CohortSelect.test.js

```diff
@@ -0,0 +1,255 @@
+    ).toBeInTheDocument();
+  });
+});
+
+test("selects all by default", () => {
+  const { getByRole } = openMenu();
+  testOptions.forEach((opt) => {
+    expect(
+      getByRole("option", { name: opt.label, selected: true })
+    ).toBeVisible();
+  });
+});
+
+test("toggles selection", () => {
+  const { getByRole } = openMenu();
+  const firstOption = getByRole("option", { name: testOptions[0].label });
+  act(() => userEvent.click(firstOption));
+  expect(firstOption.getAttribute("aria-selected")).toBe("false");
+  const menuButton = getByRole("button");
+  expect(menuButton).toHaveTextContent(
+    `${testOptions[1].label} and ${testOptions.length - 2} others`
+  );
+  act(() => userEvent.click(firstOption));
+  expect(firstOption.getAttribute("aria-selected")).toBe("true");
+  expect(menuButton).toHaveTextContent(
+    `${testOptions[0].label} and ${testOptions.length - 1} others`
+  );
+});
+
+test("toggles selection on mobile", () => {
+  // hook returns true on mobile
+  useBreakpoint.mockReturnValue(true);
+
+  const { getByRole, getByText } = render(
+    <CohortSelect
+      onChange={mockOnChange}
+      onHighlight={mockOnHighlight}
+      options={testOptions}
+    />
+  );
+  const menu = getByRole("listbox", { name: /cohort/i });
+  const valueEl = getByText((content, element) => {
+    // this pattern is permissive about spaces between words because textContent
+    // concatenated from arbitrary children may have random line breaks, etc
+    return new RegExp(
+      String.raw`^${testOptions[0].label}\s+and\s+${
+        testOptions.length - 1
+      }\s+others$`
+    ).test(element.textContent);
+  });
+
+  act(() => {
+    userEvent.deselectOptions(menu, testOptions[0].id);
+  });
+  expect(valueEl).toHaveTextContent(
+    `${testOptions[1].label} and ${testOptions.length - 2} others`
+  );
+  act(() => {
+    userEvent.selectOptions(menu, testOptions[0].id);
+  });
+  expect(valueEl).toHaveTextContent(
+    `${testOptions[0].label} and ${testOptions.length - 1} others`
+  );
+});
+
+test("sends initial selection to callback", () => {
+  render(
+    <CohortSelect
+      onChange={mockOnChange}
+      onHighlight={mockOnHighlight}
+      options={testOptions}
+    />
+  );
+  expect(mockOnChange.mock.calls.length).toBe(1);
+  // pass only the selected IDs, not the entire options object
+  expect(mockOnChange.mock.calls[0][0]).toEqual(testOptions);
+});
+
+test("sends updated selections to callback", () => {
+  const { getByRole } = openMenu();
+  const firstOption = getByRole("option", { name: testOptions[0].label });
+  act(() => userEvent.click(firstOption));
+  expect(mockOnChange.mock.calls[1][0]).toEqual(testOptions.slice(1));
+});
+
+test("applies colors to selected items", () => {
+  const { getByRole } = openMenu();
+  testOptions.forEach((opt) => {
+    expect(
+      getByRole("option", { name: opt.label, selected: true })
+    ).toHaveStyle(`background-color: ${opt.color}`);
```

Comment:
It might be helpful to give each of the testOptions a different color to make this assertion little stronger.